### PR TITLE
fix: acknowledge written

### PR DESCRIPTION
**Pull request by tbplong**
acknowledge written, added position to insert picture and picture included in the same folder.

### CODE REVIEW

File: research/acknownledge.txt [hunk 1]
```diff
@@ -1 +1,75 @@
...
+- Stock Size Constraint: The cuts must fit within the dimensions of the stock sheet.
+- Demand Constraint: Each demanded piece must be produced in the required quantity.
+- Non-Overlap Constraint: The pieces cannot overlap on the stock sheet; each piece must occupy a unique position.
+- Cutting Pattern Constraint: Often, only certain types of cuts are allowed 
+  Types of Cutting Patterns
+  + Guillotine Cuts: Each cut must go from one edge of the material to the other, simplifying the cutting process. Guillotine cutting is often preferred because it's easier to automate and more practical in large-scale manufacturing.
+
+  + Non-Guillotine Cuts: Cuts don’t have to span the entire length or width, which can lead to more efficient use of material but is usually harder to implement due to increased cutting complexity.
+
+Approaches to Solving 2D CSP
```
Comment (GitGud031005) [inline research/acknownledge.txt:27]: ? Later in algorithms

File: research/acknownledge.txt [hunk 2]
```diff
@@ -1 +1,75 @@
...
+Solution Steps
+- Define Patterns: Determine potential patterns for arranging the pieces on the stock sheet. Patterns should be as efficient as possible (e.g., filling rows or columns).
+
+- Select Patterns: Using an algorithm (e.g., Best Fit), place pieces in each selected pattern until the demand is met or no further patterns can be efficiently placed.
+
+- Minimize Waste: After meeting all demands, calculate the leftover area to assess efficiency.
+
+############(Insert Screenshot_139.webp included in same folder).
+
+Applications of 2D CSP
```
Comment (GitGud031005) [inline research/acknownledge.txt:69]: Application rep in intro

File: research/acknownledge.txt [hunk 3]
```diff
@@ -1 +1,75 @@
...
+
+3. Global Methods
+Global methods strive to find the absolute best (optimal) solution by exhaustively evaluating possible configurations. They often use mathematical formulations, like integer programming (IP), combined with techniques such as:
+
+Integer Programming (IP): Models the problem with binary variables and constraints, ensuring each piece is accounted for in the most efficient arrangement.
+Column Generation: Dynamically generates cutting patterns to improve material utilization and selects the best set of patterns.
+
+Global methods ensure minimal waste but are computationally intensive and are best suited for high-value materials or stable demand scenarios where the effort of obtaining the optimal layout is justified.
+
+Example Problem
```
Comment (GitGud031005) [inline research/acknownledge.txt:54]: Example should show difficulty, the main problem to the 2DCSP

File: research/acknownledge.txt [hunk 4]
```diff
@@ -1 +1,75 @@
...
+Two pieces of size 3 x 3
+Four pieces of size 2 x 2
+Solution Steps
+- Define Patterns: Determine potential patterns for arranging the pieces on the stock sheet. Patterns should be as efficient as possible (e.g., filling rows or columns).
+
+- Select Patterns: Using an algorithm (e.g., Best Fit), place pieces in each selected pattern until the demand is met or no further patterns can be efficiently placed.
+
+- Minimize Waste: After meeting all demands, calculate the leftover area to assess efficiency.
+
+############(Insert Screenshot_139.webp included in same folder).
```
Comment (GitGud031005) [inline research/acknownledge.txt:67]: Excel block size width and length not equal, not suitable for demonstrating

**Review by GitGud031005 — CHANGES_REQUESTED** [review]

File: research/acknownledge.txt (hunk 1 above)
Comment (tbplong) [reply research/acknownledge.txt:27]: Removed

File: research/acknownledge.txt (hunk 2 above)
Comment (tbplong) [reply research/acknownledge.txt:69]: Remove

File: research/acknownledge.txt (hunk 3 above)
Comment (tbplong) [reply research/acknownledge.txt:54]: New example given

File: research/acknownledge.txt (hunk 4 above)
Comment (tbplong) [reply research/acknownledge.txt:67]: Image will be provided later

File: research/acknownledge.txt [hunk 5]
```diff
@@ -1 +1,47 @@
-Author: Chau Anh Nhat - 2352856
```
Comment (GitGud031005) [inline research/acknownledge.txt:1]: Author name here

File: research/acknownledge.txt [hunk 6]
```diff
@@ -1 +1,47 @@
...
+Six pieces of size 15x10 inches
+Five pieces of size 10x10 inches
+Four pieces of size 12x15 inches
+Eight pieces of size 6x8 inches
+Ten pieces of size 5x5 inches
+
+Difficulties in Solving this Example
+- Piece Size and Shape Diversity:
+  + The pieces have different dimensions, some long and narrow, others square or almost square. This variety complicates the arrangement, as pieces cannot       easily “tessellate” or fit together like a puzzle without leaving gaps.
+  + Large pieces, such as 20x10, reduce flexibility in the arrangement, limiting the space left for smaller pieces.
```
Comment (GitGud031005) [inline research/acknownledge.txt:36]: Typing error

File: research/acknownledge.txt [hunk 7]
```diff
@@ -1 +1,47 @@
...
+
+Example Problem
+Imagine a factory is cutting a large metal sheet that is 80 inches by 50 inches. The factory needs to produce a range of parts with specific sizes to minimize waste while meeting various order requirements. The parts required are:
+
+Six pieces of size 15x10 inches
+Five pieces of size 10x10 inches
+Four pieces of size 12x15 inches
+Eight pieces of size 6x8 inches
+Ten pieces of size 5x5 inches
+
```
Comment (GitGud031005) [inline research/acknownledge.txt:32]: Provide images showing a way to solve

Comment (tbplong) [reply research/acknownledge.txt:32]: This should be done later on, when we have constructed the ILP model or discuss about an exact solution.

File: research/acknownledge.txt (hunk 6 above)
Comment (tbplong) [reply research/acknownledge.txt:36]: fixed, changed from 20x10 to 15x10, if there's any other error let me know

**Comment by GitGud031005** [comment]
Should be reconsidered after understanding modeling and solution

**Pull request closed without merging**